applications/openshift/api-server/api_server_admission_control_plugin_SecurityContextDeny/rule.yml

```diff
@@ -19,22 +23,56 @@ description: |-
           kind: DefaultAdmissionConfig
           apiVersion: v1
           disable: false</pre>
+{{%- endif %}}
+    
 
 rationale: |-
+{{%- if product == "ocp4" %}}
+    SecurityContextDeny can be used to provide a layer of security for clusters which do not have PodSecurityPolicies enabled.
+{{% else %}}
     Setting admission control policy to <tt>SecurityContextDeny</tt> denies the
     pod level SecurityContext customization. Any attempts to customize the
     SecurityContext that are not explicitly defined in the Pod Security Policy
     (PSP) are blocked. This ensures that all pods adhere to the PSP defined
     by your organization and you have a uniform pod level security posture.
+{{%- endif %}}
 
 severity: medium
 
 references:
     cis: 1.2.13
 
-ocil_clause: '<tt>admissionConfig</tt> does not contain <tt>SecurityContextDeny</tt>'
+ocil_clause: |-
+{{%- if product == "ocp4" %}}
+    '<tt>enable-admission-plugins</tt>does not contain <tt>SecurityContextConstraint</tt>'
+{{% else %}}
+    '<tt>admissionConfig</tt> does not contain <tt>SecurityContextDeny</tt>'
+{{%- endif %}}
 
 ocil: |-
+{{%- if product == "ocp4" %}}
+    The SecurityContextConstraint plugin should be enabled in the list of enabled plugins in the apiserver configuration:
+    <pre>$ oc get configmap config -n openshift-kube-apiserver -ojson | jq -r '.data["config.yaml"]' | jq '.apiServerArguments."enable-admission-plugins"' | grep 'SecurityContextConstraint' </pre>
+{{% else %}}
     Run the following command on the master node(s):
     <pre>$ sudo grep -A4 SecurityContextDeny /etc/origin/master/master-config.yaml</pre>
     The output should return <pre>disable: false</pre>.
+{{%- endif %}}
+
+{{%- if product == "ocp4" %}}
+warnings:
+- general: |-
+    {{{ openshift_cluster_setting("/api/v1/namespaces/openshift-kube-apiserver/configmaps/config") | indent(4) }}}
+{{%- endif %}}
+
+template:
+  name: yamlfile_value
+  vars:
+    ocp_data: "true"
+    entity_check: "none satisfy"
+    filepath: /api/v1/namespaces/openshift-kube-apiserver/configmaps/config
+    yamlpath: '.data["config.yaml"]'
+    values:
+    - value: '"enable-admission-plugins":\[[^]]*"SecurityContextConstraint"'
```

Comment:
This one is a little confusing to me. Isn't this about SecurityContextDeny?

Reply:
If my understanding of the documentation is correct, For 4.6 and above, verify that the set of admission plugins includes SecurityContextConstraint, is that correct ?

Reply:
This rule is about SecurityContextDeny and not SecurityContextConstraint. SecurityContextConstraint rules are found under `applications/openshift/scc`. If we need to verify that SecurityContextConstraint is enabled, we need a new rule.

Reply:
Fixed!